classes/local/table_manager.php

```diff
@@ -51,14 +53,22 @@ public static function get_lockout_table() {
             get_string('username'),
             get_string('email'),
             get_string('fullname'),
+            get_string('formlockouttier', 'tool_securityquestions'),
             get_string('actions'),
         );
         $table->attributes['class'] = 'generaltable table table-bordered';
-        $table->colclasses = array('centeralign', 'centeralign', 'centeralign', 'centeralign', 'centeralign');
+        $table->align = array('center', 'left', 'left', 'left', 'center', 'left');
 
         foreach ($lockedusers as $userrecord) {
             $user = $DB->get_record('user', array('id' => $userrecord->userid));
 
+            // Initialise the lock, to check for expired locks.
+            tool_securityquestions_initialise_lockout_counter($user);
+            // If lock was reset to 0, ignore this user.
+            if ($DB->get_field('tool_securityquestions_loc', 'tier', array('userid' => $user->id)) == 0) {
```

Comment:
is this dead code? it is filtered above

Reply:
So this just calls the initialiser, which will clear the lock if it is expired, and if it was cleared, it skips displaying this user, as they are no longer locked. This function probably needs a rename, to make it clear exactly what it does, after adding the expiry functionality

Reply:
Renamed the function to refresh_lock_state, which is a little clearer what is actually being done when it's called